chi_edge/image.py

```diff
@@ -0,0 +1,70 @@
+"""Utilities for reading and writing to disk image."""
+from pathlib import Path
+
+from chi_edge.vendor.FATtools import Volume
+from chi_edge.vendor.FATtools.partutils import partition
+
+
+def printn(s):
+    print(s)
+
+
+def find_boot_partition_id(image: str):
+
+    # max of 128 gpt partitions possible
+    for partid in range(0, 128):
+        try:
+            part = Volume.vopen(
+                image,
+                mode="r+b",
+                what=f"partition{partid}",
+            )
+            partfs = Volume.openvolume(part)
+        except ZeroDivisionError:
```

Comment:
lol, how can this happen here?

Reply:
As it iterates over the 16 different special nvidia partitions, some subset make it very unhappy when calculating buffer sizes and similar things.